filebeat/input/filestream/prospector.go

```diff
@@ -144,11 +219,11 @@ func (p *fileProspector) Init(
 		}
 
 		// Return early (do nothing) if:
-		//  - The identifiers are the same
-		//  - The old identifier is neither native nor path
+		//  - The old identifier is neither native, path or fingerprint
 		oldIdentifierName := fm.IdentifierName
-		if oldIdentifierName == identifierName ||
-			!(oldIdentifierName == nativeName || oldIdentifierName == pathName) {
+		if !(oldIdentifierName == nativeName ||
+			oldIdentifierName == pathName ||
+			oldIdentifierName == fingerprintName) {
```

Comment:
Took me a second to see the `!`, perhaps the inverted condition would be easier to read?

```suggestion
		if oldIdentifierName != nativeName &&
			oldIdentifierName != pathName &&
			oldIdentifierName != fingerprintName {
```

Reply:
Maybe, I'll update to your suggestion.